### PR TITLE
Removes integer suffix from "datagroup" label in SPIO output

### PR DESCRIPTION
**Pull request by nselliott**
# Summary

- This PR is an enhancement.
- It does the following:
  - Removes the "_%07d" suffix from the outer "datagroup" in output files when using M->M file per processor output.
  - Addresses https://github.com/LLNL/axom/issues/386
  - This will make future enhancements to automatic blueprint index generation easier, and this limited change doesn't break current usage in applications.

### CODE REVIEW

File: src/axom/sidre/spio/IOManager.cpp [hunk 1]
```diff
@@ -216,7 +216,15 @@ void IOManager::write(sidre::Group* datagroup,
     }
     SLIC_ASSERT(h5_file_id >= 0);
 
-    std::string group_name = fmt::sprintf("datagroup_%07d", m_my_rank);
+    std::string group_name;
```
Comment (agcapps) [inline src/axom/sidre/spio/IOManager.cpp:219]: Style question: Why not initialize `group_name = "datagroup"` like the other changes?

**Review by agcapps — APPROVED** [review]
Thanks, @nselliott .  My only question was a style point.

File: src/axom/sidre/spio/IOManager.cpp (hunk 1 above)
Comment (rhornung67) [reply src/axom/sidre/spio/IOManager.cpp:219]: I agree that it should be initialized on construction, if possible. @nselliott is there a reasonable default?

Comment (nselliott) [reply src/axom/sidre/spio/IOManager.cpp:219]: Yes, this is an inconsistency that I missed.  I will address it.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**